### ISSUE TITLE
[RFC]: optee-os build is broken with gcc10

### ISSUE THREAD

**Issue by zandrey** [Freescale/meta-freescale#407]
Currently, the build of `optee-os` breaks on `master` which is relateed to the gcc10 upgrade. Following build error is reported:

```
aarch64-fsl-linux-ld.bfd: /development/yocto-master/build-output/work/imx8mmevk-fsl-linux/optee-os/3.2.0.imx-r0/recipe-sysroot/usr/lib/aarch64-fsl-linux/10.1.0/libgcc.a(lse-init.o): in function `init_have_lse_atomics':
  | /usr/src/debug/libgcc/10.1.0-r0/gcc-10.1.0/build.aarch64-fsl-linux.aarch64-fsl-linux/libgcc/../../../../../../../work-shared/gcc-10.1.0-r0/gcc-10.1.0/libgcc/config/aarch64/lse-init.c:44: undefined reference to `__getauxval'
  | core/arch/arm/kernel/link.mk:36: recipe for target '/development/yocto-master/build-output/work/imx8mmevk-fsl-linux/optee-os/3.2.0.imx-r0/build.mx8mmevk/core/all_objs.o' failed
  | make: *** [/development/yocto-master/build-output/work/imx8mmevk-fsl-linux/optee-os/3.2.0.imx-r0/build.mx8mmevk/core/all_objs.o] Error 1
  | make: *** Waiting for unfinished jobs....
  | make: Leaving directory '/development/yocto-master/build-output/work/imx8mmevk-fsl-linux/optee-os/3.2.0.imx-r0/git'
  | WARNING: /development/yocto-master/build-output/work/imx8mmevk-fsl-linux/optee-os/3.2.0.imx-r0/temp/run.do_compile.8135:1 exit 1 from 'exit 1'
```
There is a potential fix for this issue to add ` -mno-outline-atomics` to `CFLAGS` in `do_compile` task, which solves the build issue.

It has been discussed in [OE-Core mailing list](https://lists.openembedded.org/g/openembedded-core/message/138437) and the conclusion is that this fix could be applied at a [(potentially) marginal performance penalty](https://lists.openembedded.org/g/openembedded-core/message/138271).

I've tried to add this flag and it does solves the build issue, but given the latest state of proposed fixes after GCC10 upgrade ([imx-gpu-sdk PR in meta-freescale-distro](https://github.com/Freescale/meta-freescale-distro/pull/31)), I'd like to take the discussion to the community to see if the fix is appropriate.

Any thoughts here?

-- andrey

**Comment by zandrey** [comment]
This fix looks pretty solid, as there a is a PR in `optee-os` upstream: [3874](https://github.com/OP-TEE/optee_os/pull/3874)

Once it get's merged - I'll backport that patch and put it in the layer here.

For the time being and those who suffers from failed build - please use the WA from the issue description.

-- andrey

**Comment by zandrey** [comment]
Upstream PR has been accepted with modifications, therefore I've opened #437 to have a proper backport added to the layer.

This issue could be closed, as OP-TEE build succeeds with #437
